tests/ut/attention_sharded_test.cpp

```diff
@@ -0,0 +1,163 @@
+    // Define the input parameters for the function
+    const int presentSeqLen = inputSeqLen + pastSeqLen;
+    const int qHeadNum = headNum;
+    const int kvHeadNum = headNum;
+    const int oStride = qHeadNum * headSize;
+    const int qStride = qHeadNum * headSize;
+    const int kvStride = kvHeadNum * headSize;
+    const int batchSize = 1;
+    const float scale = 1 / sqrtf(headSize);
+
+    // Set OpenMP threads
+    omp_set_num_threads(threadNum);
+
+    // Define the input and output arrays
+    bfloat16_t *query = (bfloat16_t *)malloc(batchSize * inputSeqLen * qHeadNum * headSize * sizeof(bfloat16_t));
+    bfloat16_t *output = (bfloat16_t *)malloc(batchSize * inputSeqLen * qHeadNum * headSize * sizeof(bfloat16_t));
+    bfloat16_t *refOut = (bfloat16_t *)malloc(batchSize * inputSeqLen * qHeadNum * headSize * sizeof(bfloat16_t));
+    float16_t *key = (float16_t *)malloc(batchSize * presentSeqLen * kvHeadNum * headSize * sizeof(float16_t));
+    float16_t *value = (float16_t *)malloc(batchSize * presentSeqLen * kvHeadNum * headSize * sizeof(float16_t));
+    float *attnMask = (float *)malloc(batchSize * inputSeqLen * presentSeqLen * sizeof(float));
+
+    // Initialize the input arrays
+    for (int i = 0; i < batchSize * inputSeqLen * qHeadNum * headSize; i++) {
+        query[i] = (bfloat16_t)((rand() % 1000 - 500) / 500.0f);
+    }
+    for (int i = 0; i < batchSize * presentSeqLen * kvHeadNum * headSize; i++) {
+        key[i] = (float16_t)((rand() % 1000 - 500) / 500.0f);
+        value[i] = (float16_t)((rand() % 1000 - 500) / 500.0f);
+    }
+
+    memset(attnMask, 0, batchSize * inputSeqLen * presentSeqLen * sizeof(float));
+
+    // Call the function under test
+    xft::crossAttnShardedHead<bfloat16_t, float16_t>(
+            output, query, attnMask, inputSeqLen, presentSeqLen, qHeadNum, headSize, oStride, qStride, batchSize, scale,
+            threadNum, [&](int b, int qHead) { return std::make_tuple(key + qHead * headSize, kvStride, nullptr); },
+            [&](int b, int qHead) { return std::make_tuple(value + qHead * headSize, kvStride, nullptr); });
+
+    for (int i = 0; i < batchSize; ++i) {
+        auto pout = refOut + i * inputSeqLen * oStride;
+        auto pquery = query + i * inputSeqLen * qStride;
+        auto pkey = key + i * presentSeqLen * kvStride;
+        auto pvalue = value + i * presentSeqLen * kvStride;
+        crossAttentionRef(pout, pquery, pkey, pvalue, attnMask, inputSeqLen, presentSeqLen, qHeadNum, headSize, scale);
+    }
+
+    // Compare values
+    for (int i = 0; i < batchSize * inputSeqLen * qHeadNum * headSize; i++) {
+        ASSERT_NEAR((float)output[i], (float)refOut[i], 1e-3);
+    }
+
+    // Clean up
+    free(query);
+    free(output);
+    free(refOut);
+    free(key);
+    free(value);
+    free(attnMask);
+}
+
+// Define a test case
+TEST_F(AttentionKernelsTest, CrossAttnShardedHeadTest1) {
```

Comment:
didn't see what the specific differences are between these two sets of test cases. Can we use a more specific naming convention for the test cases? Also, since we already support the int8 kvcache format, should it be reflected in this unit test?

Reply:
will modify and create another PR.